batch/batch/worker.py

```diff
@@ -116,10 +116,20 @@ def docker_call_retry(timeout, name):
 async def create_container(config, name):
     delay = 0.1
     error = 0
+
     while True:
         try:
             return await docker.containers.create(config, name=name)
         except DockerError as e:
+            async def handle_error():
+                nonlocal error, delay
+                error += 1
+                if error < 10:
+                    delay = await sleep_and_backoff(delay)
+                    return True
+                log.exception(f'encountered 10 errors while creating container {name}, aborting', stack_info=True)
+                return False
+
```

Comment:
Since this doesn't depend on `e` at all, let's move it outside the try and the while loop, right under delay and error.